raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -702,27 +657,50 @@ private void updateVoterAndObserverStates(VoterSet lastVoterSet) {
 
         // Move any of the remaining old voters to observerStates
         for (ReplicaState replicaStateEntry : oldVoterStates.values()) {
+            replicaStateEntry.clearListeners();
             observerStates.putIfAbsent(replicaStateEntry.replicaKey, replicaStateEntry);
         }
     }
 
-    private static class ReplicaState implements Comparable<ReplicaState> {
-        ReplicaKey replicaKey;
-        Optional<LogOffsetMetadata> endOffset;
-        long lastFetchTimestamp;
-        long lastFetchLeaderLogEndOffset;
-        long lastCaughtUpTimestamp;
-        boolean hasAcknowledgedLeader;
+    static class ReplicaState implements Comparable<ReplicaState> {
```

Comment:
This static class should be public as it is exposed in `RaftUtil.singletonDescribeQuorumResponse`. Right now this class is exposed outside of it's visibility scope.